composer.json

```diff
@@ -22,10 +22,10 @@
     "zendframework/zend-diactoros": "~1.1",
     "container-interop/container-interop": "^1.1",
     "zendframework/zend-math": "~2.2",
-    "zendframework/zend-eventmanager": "~2.2"
+    "zendframework/zend-eventmanager": "~2.2 || ~3.0"
```

Comment:
Actually after some thoughts, event manager is not needed at all in middleware world. The current usage of EventManager was to trigger an event whenever a token is created.

But in middleware this is not needed: you just need to register a middleware just after the main middleware (either at the end of pipeline...) and alter the response. Therefore you can remove this dependency and all usage of event manager in the code :).